homeassistant/components/metoffice/data.py

```diff
@@ -0,0 +1,87 @@
+"""Common Met Office Data class used by both sensor and entity."""
+
+import logging
+
+import datapoint
+
+from .const import MODE_3HOURLY
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MetOfficeData:
+    """Get current and forecast data from Datapoint.
+
+    Please note that the 'datapoint' library is not asyncio-friendly, so some
+    calls have had to be wrapped with the standard hassio helper
+    async_add_executor_job.
+    """
+
+    def __init__(self, hass, api_key, latitude, longitude):
+        """Initialize the data object."""
+        self._hass = hass
+        self._latitude = latitude
+        self._longitude = longitude
+
+        self._datapoint = datapoint.connection(api_key=api_key)
+        self._site = None
+
+        # Holds the current data from the Met Office
+        self.site_id = None
+        self.site_name = None
+        self.now = None
+
+    @property
+    def latitude(self):
```

Comment:
We don't need these properties since they are just returning instance attributes. Just use regular instance attributes.

Reply:
Right-o, fair enough.